pkg/operator/operator_config.go

```diff
@@ -223,8 +475,11 @@ func (r *operatorConfigReconciler) ensureRuleEvaluatorSecrets(ctx context.Contex
 func (r *operatorConfigReconciler) ensureRuleEvaluatorDeployment(ctx context.Context, rules *monitoringv1alpha1.RuleEvaluatorSpec) error {
 	deploy := r.makeRuleEvaluatorDeployment(rules)
 
-	// Upsert rule-evaluator ConfigMap.
-	if err := r.client.Update(ctx, deploy); err != nil {
+	// Upsert rule-evaluator Deployment.
+	// We've to use Patch() as Update() with update we'll get stuck in an infinite loop as each
+	// update increases the generation since we override the managed fields metadata, which is
+	// set by the kube-controller-manager, which wants to own an annotation on deployments.
+	if err := r.client.Patch(ctx, deploy, client.Merge); apierrors.IsNotFound(err) {
```

Comment:
This here only became evident after changing Owns to Watches in the setup so that changes to the deployment actually triggered reconciliation. We could certainly consider patching everywhere in the future but for now I just wanted to get this to work.

Reply:
Makes sense, thanks for explaining.